In [254]:
import pandas as pd

## importing the dataset

In [255]:
data = pd.read_csv("train.csv")

## EDA

In [256]:
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [257]:
data.drop(["PassengerId","Name","Ticket","Cabin"],axis=1,inplace=True)

In [258]:
data.head()

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0         0       3    male  22.0      1      0   7.2500        S
1         1       1  female  38.0      1      0  71.2833        C
2         1       3  female  26.0      0      0   7.9250        S
3         1       1  female  35.0      1      0  53.1000        S
4         0       3    male  35.0      0      0   8.0500        S

In [259]:
data.dropna(inplace=True)

In [260]:
from sklearn import preprocessing

In [261]:
from sklearn.model_selection import train_test_split

In [262]:
from sklearn.naive_bayes import GaussianNB

In [263]:
from sklearn.metrics import accuracy_score

In [264]:
from sklearn.metrics import confusion_matrix

In [265]:
le = preprocessing.LabelEncoder()

In [266]:
data["Sex"]=le.fit(data["Sex"]).transform(data["Sex"])

In [267]:
data["Embarked"]=le.fit(data["Embarked"]).transform(data["Embarked"])

In [365]:
data.head()

Survived  Pclass  Sex  Age  SibSp  Parch     Fare  Embarked
0         0       3    1  0.0      1      0   7.2500         2
1         1       1    0  1.0      1      0  71.2833         0
2         1       3    0  1.0      0      0   7.9250         2
3         1       1    0  1.0      1      0  53.1000         2
4         0       3    1  0.0      0      0   8.0500         2

## creating the model for Pclass

In [282]:
y=data.Pclass

In [283]:
y

0      3
1      1
2      3
3      1
4      3
      ..
884    2
885    1
886    3
887    1
888    3
Name: Pclass, Length: 889, dtype: int64

In [284]:
X=data.drop("Pclass",axis=1)

In [285]:
X

Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0           0    1  22.0      1      0   7.2500         2
1           1    0  38.0      1      0  71.2833         0
2           1    0  26.0      0      0   7.9250         2
3           1    0  35.0      1      0  53.1000         2
4           0    1  35.0      0      0   8.0500         2
..        ...  ...   ...    ...    ...      ...       ...
884         0    1  27.0      0      0  13.0000         2
885         1    0  19.0      0      0  30.0000         2
886         0    0  45.0      1      2  23.4500         2
887         1    1  26.0      0      0  30.0000         0
888         0    1  32.0      0      0   7.7500         1

[889 rows x 7 columns]

In [286]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [288]:
from sklearn.naive_bayes import *

In [289]:
clf = BernoulliNB()

In [290]:
y_pred = clf.fit(X_train,y_train).predict(X_test)

In [291]:
accuracy_score(y_test,y_pred,normalize=True)

0.5917602996254682

In [292]:
confusion_matrix(y_test,y_pred)

array([[ 30,   8,  32],
       [  9,  10,  30],
       [ 24,   6, 118]], dtype=int64)

### **the accuracy is low because only 158 records are perfectly fit in the model**

## creating the model for Sex


In [303]:
data.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

In [332]:
data["Age"]=data.round(data["Age"])

In [322]:
y=data["Age"]

In [323]:
X=data.drop("Age",axis=1)

In [324]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [325]:
from sklearn.naive_bayes import *

In [326]:
clf = BernoulliNB()

In [327]:
y_pred = clf.fit(X_train,y_train).predict(X_test)

In [328]:
accuracy_score(y_test,y_pred,normalize=True)

1.0

In [329]:
confusion_matrix(y_test,y_pred)

array([[157,   0],
       [  0, 110]], dtype=int64)

### **as there are no improperly fit records that's why the accuarcy is 1**

## creating the model for SibSp


In [330]:
data.head()

Survived  Pclass  Sex  Age  SibSp  Parch     Fare  Embarked
0         0       3    1  0.0      1      0   7.2500         2
1         1       1    0  1.0      1      0  71.2833         0
2         1       3    0  1.0      0      0   7.9250         2
3         1       1    0  1.0      1      0  53.1000         2
4         0       3    1  0.0      0      0   8.0500         2

In [333]:
y=data["SibSp"]

In [342]:
X=data.drop(["SibSp"],axis=1)

In [343]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [344]:
from sklearn.naive_bayes import *

In [345]:
clf = BernoulliNB()

In [346]:
y_pred = clf.fit(X_train,y_train).predict(X_test)

In [347]:
accuracy_score(y_test,y_pred,normalize=True)

0.6329588014981273

### **the accuracy of the model is 0.63**

## creating the model for Parch


In [349]:
y=data["Parch"]

In [350]:
X=data.drop("Parch",axis=1)

In [351]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [352]:
from sklearn.naive_bayes import *

In [353]:
clf = BernoulliNB()

In [354]:
y_pred = clf.fit(X_train,y_train).predict(X_test)

In [355]:
accuracy_score(y_test,y_pred,normalize=True)

0.7191011235955056

### **the accuracy of the model is 0.719**

## creating the model for Embarked


In [356]:
y=data["Embarked"]

In [358]:
X=data.drop("Embarked",axis=1)

In [359]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [360]:
from sklearn.naive_bayes import *

In [361]:
clf = BernoulliNB()

In [362]:
y_pred = clf.fit(X_train,y_train).predict(X_test)

In [363]:
accuracy_score(y_test,y_pred,normalize=True)

0.7228464419475655

### **the accuracy of the model is 0.722**